client/me/purchases/cancel-purchase/button.jsx

```diff
@@ -52,6 +53,9 @@ const CancelPurchaseButton = React.createClass( {
 						}
 					}
 				), { persistent: true } );
+
+				clearPurchases();
```

Comment:
I wonder if we shouldn't invalidate purchases right from the [`cancelPurchase` action](https://github.com/Automattic/wp-calypso/blob/master/client/lib/upgrades/actions/purchases.js#L21) instead.